lightwood/api/ensemble.py

```diff
@@ -0,0 +1,64 @@
+from .predictor import Predictor
+from lightwood.constants.lightwood import ColumnDataTypes
+from collections import Counter
+import numpy as np
+import pickle
+import torch
+import os
+
+
+class LightwoodEnsemble:
+    def __init__(self, predictor=None, predictors=None, load_from_path=None):
+        self.path_list = None
+        if load_from_path is not None:
+            with open(os.path.join(load_from_path, 'lightwood_data'), 'rb') as pickle_in:
+                obj = pickle.load(pickle_in)
+                self.path = load_from_path
+                self.path_list = obj.path_list
+                self.ensemble = [Predictor(load_from_path=path) for path in self.path_list]
+        elif predictor is not None:
+            self.ensemble = [predictor]
+        elif isinstance(predictors, list):
+            self.ensemble = predictors
+
+    def append(self, predictor):
+        if isinstance(self.ensemble, list):
+            self.ensemble.append(predictor)
+        else:
+            self.ensemble = [predictor]
+
+    def __iter__(self):
+        yield self.ensemble
+
+    def predict(self, when_data):
+        predictions = [p.predict(when_data=when_data) for p in self.ensemble]
+        formatted_predictions = {}
+        for target in self.ensemble[0].config['output_features']:
+            target_name = target['name']
+            formatted_predictions[target_name] = {}
+            pred_arr = np.array([p[target_name]['predictions'] for p in predictions])
+            if target['type'] == ColumnDataTypes.NUMERIC:
+                final_preds = np.mean(pred_arr, axis=0).tolist()
+            elif target['type'] == ColumnDataTypes.CATEGORICAL:
+                final_preds = []
+                for idx in range(pred_arr.shape[1]):
+                    final_preds.append(max(Counter(pred_arr[:, idx])))
+
+            formatted_predictions[target_name]['predictions'] = final_preds
```

Comment:
If data type is neither categorical nor numeric, this line will raise `NameError`
Solution is to add an `else: raise Exception('only numeric and categorical datatypes are supported for ensembles')` above  in the `if else`